tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -35,9 +35,9 @@ class DistributedTransformerTest
 
   void SetUp() {
     MultiDeviceTest::SetUp();
-    if (H % D != 0) {
+    if ((4 * E) % D != 0) {
```

Comment:
Is this required for all the tests here or just the MLP tests? I think MHA is fine without this requirement.

Reply:
Hmm just MLP, I'll move it there